### PR TITLE
Specify party when signing, cancelling, and attesting

### PR DESCRIPTION
**Pull request by silasdavis**
Signed-off-by: Silas Davis <silas@monax.io>

### CODE REVIEW

File: src/agreements/AgreementsAPI.sol [hunk 1]
```diff
@@ -86,4 +86,52 @@ library AgreementsAPI {
...
+  /**
+   * @dev Evaluates the possible actors of msg.sender and tx.origin against the given agreement to determine if they
+   * may act as party.
+   * @param _agreementAddress an ActiveAgreement address
+   * @param _party the party to check
+   * @return party - the agreement party associated with the identified actor. This is typically the same as the actor, but can also contain
+   * @return actor - the address of either msg.sender or tx.origin depending on which one was authorized; 0x0 if authorization failed
+   * an Organization address if an Organization was registered as a party. 0x0 if authorization failed
+   */
+  function authorizeActorAsParty(address _agreementAddress, address _party) public returns (address actor) {
```
Comment (silasdavis) [inline src/agreements/AgreementsAPI.sol:98]: FAO: reviewer

File: src/agreements/DefaultActiveAgreement.sol [hunk 2]
```diff
@@ -14,278 +14,315 @@ import "agreements/AbstractActiveAgreement_v1_0_1.sol";
...
+    ErrorsLib.revertIf(signee == address(0),
+      ErrorsLib.UNAUTHORIZED(), "DefaultActiveAgreement.sign()", "The caller is not authorized to sign");
+
+    executeSign(signee, party);
+  }
+
+  /**
+  * @dev sign using the current actor (msg.sender/tx.origin) attemping to authorize as the provided party
+  */
+  function signAsParty(address _party) external {
```
Comment (silasdavis) [inline src/agreements/DefaultActiveAgreement.sol:151]: FAO: reviewer

File: src/agreements/DefaultActiveAgreement.sol [hunk 3]
```diff
@@ -14,278 +14,315 @@ import "agreements/AbstractActiveAgreement_v1_0_1.sol";
...
+    ErrorsLib.revertIf(actor == address(0), ErrorsLib.UNAUTHORIZED(),
+      "DefaultActiveAgreement.doCancel()", "The caller is not authorized to cancel");
+
+    executeCancel(actor, party);
+  }
+
+  /**
+  * @dev cancel using the current actor (msg.sender/tx.origin) attemping to authorize as the provided party
+  */
+  function cancelAsParty(address _party) external {
```
Comment (silasdavis) [inline src/agreements/DefaultActiveAgreement.sol:271]: FAO: reviewer

File: src/lib/contracts.ts [hunk 4]
```diff
@@ -194,9 +194,10 @@ export class Contracts {
     return this.manager.ActiveAgreementRegistry.addAgreementToCollection(DecodeHex(collectionId), agreement);
   }
 
-  async signAgreement(actingUserAddress: string, agreementAddress: string) {
+  async signAgreement(actingUserAddress: string, agreementAddress: string, partyAddress?: string) {
```
Comment (silasdavis) [inline src/lib/contracts.ts:197]: FAO: reviewer (and changes in this file)

**Comment by silasdavis** [comment]
Sorry about the linting noise on this, I've marked the key entry points for the review above

File: src/agreements/Completables.sol [hunk 5]
```diff
@@ -8,328 +8,379 @@ import "commons-management/AbstractVersionedArtifact.sol";
...
+
+    return executeAttest(intervalId, timestamp, actor, party);
+  }
+
+  /**
+   * @notice attest to this this completable, deriving the franchisee from the caller via the agreement associated with
+   * the completable - attempting to attest as the specified party.
+   * @return whether the Completable is ratified after this invocation
+   */
+  function attestAsParty(bytes32 intervalId, int timestamp, address party)
```
Comment (silasdavis) [inline src/agreements/Completables.sol:258]: FAO: reviewer

**Review by ommish — APPROVED** [review]
lgtm!

**Pull request merged**